frontend/src/components/waitlist/WaitlistPage.tsx

```diff
@@ -0,0 +1,223 @@
+import { useLocalization } from "@fluent/react";
+import styles from "./WaitlistPage.module.scss";
+import { Layout } from "../layout/Layout";
+import { Button } from "../Button";
+import { FormEventHandler, ReactNode, useEffect, useState } from "react";
+import { getLocale } from "../../functions/getLocale";
+import { toast } from "react-toastify";
+import { CountryPicker } from "./CountryPicker";
+import { useRuntimeData } from "../../hooks/api/runtimeData";
+import { LocalePicker } from "./LocalePicker";
+import { useUsers } from "../../hooks/api/user";
+
+export type Props = {
+  supportedLocales: string[];
+  newsletterId: string;
+  headline: string;
+  lead: string;
+  legalese: ReactNode;
+};
+
+export const WaitlistPage = (props: Props) => {
+  const { l10n } = useLocalization();
+  const currentLocale = getLocale(l10n);
+  const runtimeData = useRuntimeData();
+  const currentCountry =
+    runtimeData.data?.PERIODICAL_PREMIUM_PLANS.country_code.toUpperCase() ??
+    (currentLocale.split("-")[1]?.toUpperCase() as string | undefined);
+  const [country, setCountry] = useState<string>();
+  const [locale, setLocale] = useState<string>(
+    props.supportedLocales.find(
+      (supportedLocale) =>
+        supportedLocale.toLowerCase() ===
+        currentLocale.split("-")[0].toLowerCase()
+    ) ?? "en"
+  );
+  const usersData = useUsers();
+  const [email, setEmail] = useState<string | undefined>(
+    usersData.data?.[0]?.email
+  );
+
+  useEffect(() => {
+    if (
+      typeof email === "undefined" &&
+      typeof usersData.data?.[0]?.email === "string"
+    ) {
+      setEmail(usersData.data[0].email);
+    }
+  }, [email, usersData]);
+
+  const onSubmit: FormEventHandler = async (event) => {
+    event.preventDefault();
+
+    if (!email) {
+      // Modern browsers should prevent this from happening, due to the `required` attribute:
+      console.error(
+        "WaitlistPage.onSubmit: form submitted while required input field was empty."
+      );
+      return;
+    }
```

Comment:
question (non blocking): If for any reason the user is able to submit the form, should the user be told as such, visually? Is this what we would use our toast notifications for?

Reply:
That's a good question. That would indeed be best, and while whether using a toast is a UI design question (but my assumption would be yes), we could definitely do something like that. That said, given [browser support for `required`](https://caniuse.com/mdn-api_htmlinputelement_required), I don't think there's that big a risk of the user being able to submit the form anyway.

And yes, I think final strings came in while I was on holiday, so I'll add those to the PR.

Reply:
OK, copy-wise the only change was casing. There's also copy for a bundle waitlist page, but I'll submit a separate PR that adds that page.